src/transformers/modeling_utils.py

```diff
@@ -1371,11 +1373,53 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                             else:
                                 raise ValueError from e
                     except (UnicodeDecodeError, ValueError):
-                        raise OSError(
-                            f"Unable to load weights from pytorch checkpoint file for '{pretrained_model_name_or_path}' "
-                            f"at '{resolved_archive_file}'. "
-                            "If you tried to load a PyTorch model from a TF 2.0 checkpoint, please set from_tf=True."
-                        )
+                        # Check sha256
+                        def sha256(filename, block_size=4096 * 1024):
+                            # Python program to find SHA256 hash string of a file
+
+                            sha256_hash = hashlib.sha256()
+                            with open(filename, "rb") as f:
+                                # Read and update hash string value in blocks of 4Mo
+                                for byte_block in iter(lambda: f.read(block_size), b""):
+                                    sha256_hash.update(byte_block)
+
+                            return sha256_hash.hexdigest()
+
+                        actual_sha = sha256(resolved_archive_file)
+                        metafilename = f"{resolved_archive_file}.json"
+                        try:
+                            with open(metafilename, "r", encoding="utf-8") as f:
+                                data = json.load(f)
+                                # For some reason etag contains quotes, we just want the sha.
```

Comment:
that's the spec => https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag

Reply:
also want to make sure people know that `etag === sha256(file)` only for LFS files

cc @Pierrci @coyotte508

Reply:
Thanks !

Reply:
BTW the etag of non-lfs files is also some kind of hash that can probably be computed client-side too: 

it's the `git-sha` of the blob, i.e. the sha1 of the string `blob SIZE\0CONTENT`

Reply:
https://stackoverflow.com/a/7225329/593036